components/Typography/src/UIFont+MaterialScalable.m

```diff
@@ -0,0 +1,98 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "UIFont+MaterialScalable.h"
+
+#import <objc/runtime.h>
+
+#import "MDCTypography.h"
+#import "private/UIContentSizeCategory+Material.h"
+
+static char MDCFontScaleObjectKey;
+
+@implementation UIFont (MaterialScalable)
+
+- (UIFont *)mdc_scaledFontForSizeCategory:(UIContentSizeCategory)sizeCategory {
+  if (!self.mdc_scalingCurve) {
+    return self;
+  }
+
+  NSNumber *fontSizeNumber;
+  if (sizeCategory) {
+    fontSizeNumber = self.mdc_scalingCurve[sizeCategory];
+  }
+
+  // Guard against broken / incomplete scaling curves by returning self if fontSizeNumber is nil.
+  if (fontSizeNumber == nil) {
+    return self;
+  }
+
+  CGFloat fontSize = (CGFloat)fontSizeNumber.doubleValue;
+
+  // Guard against broken scaling curves encoded with 0.0 or negative values
+  if (fontSize <= 0.0) {
+    return self;
+  }
+
+  UIFont *scaledFont = [UIFont fontWithDescriptor:self.fontDescriptor size:fontSize];
+  scaledFont.mdc_scalingCurve = self.mdc_scalingCurve;
+
+  return scaledFont;
+}
+
+- (UIFont *)mdc_scaledFontForCurrentSizeCategory {
+  UIContentSizeCategory currentSizeCategory = GetCurrentSizeCategory();
+
+  return [self mdc_scaledFontForSizeCategory:currentSizeCategory];
+}
+
+- (nonnull UIFont *)mdc_scaledFontAtDefaultSize {
+  return [self mdc_scaledFontForSizeCategory:UIContentSizeCategoryLarge];
+}
+
+- (CGFloat)mdc_scaledValueForValue:(CGFloat)value {
```

Comment:
This looks identical to the code in MDCFontScaler. Since that class depends on this class, please reuse this function instead.  Since it's a public API, there's no reason to avoid using it.

Reply:
MDCFontScaler does not keep instances of fonts.
I'm not sure it makes sense to do so, just to expose this functionality.

Reply:
Nuked.